client/src/pages/settings/settings.page.js

```diff
@@ -22,17 +22,173 @@ import FormControlLabel from "@mui/material/FormControlLabel";
 import { Link as RouterLink } from "react-router-dom";
 import Button from "@mui/material/Button";
 import Checkbox from "@mui/material/Checkbox";
+import useAuth from "../../hooks/useAuth";
+import bcrypt from "bcryptjs";
 
 const SettingsPage = () => {
+    const { auth, setAuth } = useAuth();
+
     const scrollOffset = -1 * window.innerHeight * 0.1;
 
-    const handleSubmit = (event) => {
+    const handleBasicsSubmit = (event) => {
         event.preventDefault();
         const data = new FormData(event.currentTarget);
-        console.log({
+
+        let json_data = JSON.stringify({
+            firstName: data.get("firstName"),
+            lastName: data.get("lastName"),
             email: data.get("email"),
-            password: data.get("password"),
         });
+        const id = auth._id;
+        console.log(id);
+        try {
+            // Send data to the backend via PUT
+            fetch(`http://localhost:8080/users/${id}`, {
+                method: "PUT",
+                headers: { "Content-Type": "application/json" },
+                body: json_data, // body data type must match "Content-Type" header
+            })
+                .then((response) => response.json())
+                .then((data) => {
+                    console.log(data);
+                })
+                .catch((error) => console.error(error));
+        } catch (err) {
+            console.error(err);
+        }
+    };
+
+    const handlePasswordSubmit = async (event) => {
+        event.preventDefault();
+        const data = new FormData(event.currentTarget);
+
+        let json_string = JSON.stringify({
+            oldPassword: data.get("old-password"),
+            newPassword: data.get("new-password"),
+            confirmPassword: data.get("confirm-password"),
+        });
+        console.log(json_string);
+        let data_obj = JSON.parse(json_string);
+        //console.log(data_obj);
+
+        const id = auth._id;
+        //console.log(id);
+        //console.log(data_obj.oldPassword, auth.password);
+
+        const isMatch = await bcrypt.compare(
+            data_obj.oldPassword,
+            auth.password
+        );
+        if (!isMatch) {
+            console.log("Old password is incorrect");
+        }
+
+        if (data_obj.confirmPassword !== data_obj.newPassword) {
+            console.log("passwords do not match");
+        } else if (data_obj.oldPassword === data_obj.newPassword) {
+            console.log("Cannot use the same password again");
+        }
+        /* INCLUDE ACTUAL ERROR HANDLING HERE/DISPLAYING THE MESSAGE */
+        const salt = await bcrypt.genSalt(10);
+        const hash = await bcrypt.hash(data.get("new-password"), salt);
+        let relevant_data = JSON.stringify({
+            password: hash,
+        });
+        try {
+            // Send data to the backend via PUT
+            fetch(`http://localhost:8080/users/${id}`, {
```

Comment:
It honestly may make more sense to store this as a separate route on the backend for resetting password and placing all that logic there? Not sure